efficientdet/keras/efficientdet_arch_keras_test.py

```diff
@@ -58,15 +58,24 @@ def test_model_output(self):
     with tf.Session(graph=tf.Graph()) as sess:
       feats = tf.ones(inputs_shape)
       tf.random.set_random_seed(SEED)
-      feats = legacy_arch.build_backbone(feats, config)
-      feats = legacy_arch.build_feature_network(feats, config)
-      feats = legacy_arch.build_class_and_box_outputs(feats, config)
+      feats = legacy_arch.efficientdet(feats, config=config)
       sess.run(tf.global_variables_initializer())
       class_output2, box_output2 = sess.run(feats)
     for i in range(3, 8):
       self.assertAllEqual(class_output1[i], class_output2[i])
       self.assertAllEqual(box_output1[i], box_output2[i])
 
+    with tf.Session(graph=tf.Graph()) as sess:
```

Comment:
@fsx950223 

I added a failing test. It turns out 

```
      feats, _ = efficientdet_arch_keras.build_backbone(feats, config)
      feats = efficientdet_arch_keras.build_feature_network(feats, config)
      feats = efficientdet_arch_keras.build_class_and_box_outputs(feats, config)
```

doesn't equal to:

```
      feats, _ = efficientdet_arch_keras.efficientdet(config)(feats)
```

I am surprised by this, because in efficientdet_arch_keras.efficientdet, it essential just calls the three function. The only difference is the "inputs = tf.keras.layers.Input([*utils.parse_image_size(config.image_size), 3])".

Any thoughts?

Reply:
It's strange, I'm diving into it.